File: src/main/java/com/dersaun/apigestaocontas/api/request/BuscarLancamentosFaturaRequestDTO.java
package com.dersaun.apigestaocontas.api.request;

import com.dersaun.apigestaocontas.domain.dtos.MesAnoDTO;
import com.dersaun.apigestaocontas.domain.dtos.PaginationDTO;
import com.dersaun.apigestaocontas.infra.utils.MesUtils;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@AllArgsConstructor
@NoArgsConstructor
@Data
public class BuscarLancamentosFaturaRequestDTO extends PaginationDTO {
    private Long idPessoa;
    private Long mes;
    private Integer ano;

    private Boolean somenteAtivos = false;

    private String searchKey;

    public MesAnoDTO getMesAno() {
        return MesUtils.getMesAnoReferencia(mes, ano);
    }
}

File: src/app/fatura/dtos/buscar-lancamentos-fatura-request.dto.ts
import { MesAnoDTO } from '@app/conta/dtos/mes-ano.dto';
import { ApiProperty } from '@nestjs/swagger';
import { AppPaginationDTO } from '@utils/dtos/pagination.dto';
import { isValidValue } from '@utils/index';
import { mesUtil } from '@utils/meses';
import { Transform } from 'class-transformer';

export class BuscarLancamentosFaturaRequestDTO extends AppPaginationDTO {
  @ApiProperty()
  @Transform(({ value }) => (isValidValue(value) ? Number(value) : value))
  idPessoa: number;

  @ApiProperty()
  @Transform(({ value }) => (isValidValue(value) ? Number(value) : value))
  idFormaPagamento: number;

  @ApiProperty()
  @Transform(({ value }) => (isValidValue(value) ? Number(value) : value))
  mes: number;

  @ApiProperty()
  @Transform(({ value }) => (isValidValue(value) ? Number(value) : value))
  ano: number;

  @ApiProperty({ default: false })
  somenteAtivos: boolean;

  @ApiProperty()
  searchKey: string;

  constructor(obj?: Partial<BuscarLancamentosFaturaRequestDTO>) {
    super();
    if (obj) {
      Object.assign(this, obj);
    }
  }

  getMesAno(): MesAnoDTO {
    return mesUtil.getMesAnoReferencia({
      mes: this.mes,
      ano: this.ano,
    });
  }
}
